roles/matrix-synapse/tasks/synapse/setup_install.yml

```diff
@@ -130,7 +130,7 @@
     mode: 0755
 
 - set_fact:
-    matrix_synapse_media_storage_providers: |
+    media_storage_providers: |
```

Comment:
Why not `matrix_synapse_media_storage_providers` here?

Reply:
I thought it needs to match `media_storage_providers:` in the `homeserver.yaml`

Reply:
The variable is part of the `matrix-synapse` role, so we prefix it with `matrix_synapse_`.

And then it's `_media_storage_providers`, because that's the configuration key that it controls.

The `matrix_synapse_media_storage_providers` variable should be defined in `defaults/main.yml` as an empty list: `matrix_synapse_media_storage_providers: []`.

People can override it manually, as well as enable various media storage providers modules (like the S3 module you're adding here). These modules will inject their configuration into the `matrix_synapse_media_storage_providers` variable.

Reply:
regardless the naming of the options.
the `media_storage_providers` stays empty

homeserver.yaml
```yaml
max_spider_size: 10M
max_upload_size: 4096M
media_storage_providers: []
media_store_path: /matrix-media-store-parent/media-store
metrics_flags: null
```